### PR TITLE
[TF FE] Support conversion of models with non-standard extensions in the path

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** It adds support of conversion models with non-standard extensions in the path. Also, now the frontend checks more carefully supported formats.

**Ticket:** 103793

### CODE REVIEW

File: tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (C) 2018-2023 Intel Corporation
+# Copyright (C) 2018-2022 Intel Corporation
```
Comment (popovaan) [inline tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py:1]: Looks like wrong change

File: src/frontends/tensorflow/src/graph_iterator_proto.hpp [hunk 2]
```diff
@@ -88,29 +88,40 @@ class GraphIteratorProto : public GraphIterator {
         }
     }
 
-    /// Set iterator to the start position
+    /// \brief Check if the input file is supported
+    template <typename T>
+    static bool is_supported(const std::basic_string<T>& path) {
+        std::ifstream pb_stream(path, std::ios::in | std::ifstream::binary);
+        auto graph_def = std::make_shared<::tensorflow::GraphDef>();
+        return pb_stream && pb_stream.is_open() && graph_def->ParsePartialFromIstream(&pb_stream);
```
Comment (gkrivor) [inline src/frontends/tensorflow/src/graph_iterator_proto.hpp:96]: Nervous place. But np if it isn't too heavy operation

**Review by gkrivor — APPROVED** [review]

File: tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py (hunk 1 above)
Comment (rkazants) [inline tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py:1]:
```suggestion
# Copyright (C) 2018-2023 Intel Corporation
```

Comment (rkazants) [reply tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py]: Fixed. Thanks)

File: src/frontends/tensorflow/src/graph_iterator_proto.hpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/graph_iterator_proto.hpp:96]: we have tests + newly added:)

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/graph_iterator_proto.hpp:96]: this is partial parsing, a sort of light parsing to check that this is compatible format.

**Review by popovaan — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

**Pull request merged**